state/store.go

```diff
@@ -187,15 +187,39 @@ func LoadValidators(db dbm.DB, height int64) (*types.ValidatorSet, error) {
 		valInfo2 := loadValidatorsInfo(db, valInfo.LastHeightChanged)
 		if valInfo2 == nil {
 			panic(
-				fmt.Sprintf(
-					"Couldn't find validators at height %d as last changed from height %d",
+				fmt.Errorf(
+					"couldn't find validators at height %d as last changed from height %d",
 					valInfo.LastHeightChanged,
 					height,
 				),
 			)
 		}
-		valInfo2.ValidatorSet.IncrementProposerPriority(int(height - valInfo.LastHeightChanged)) // mutate
 		valInfo = valInfo2
```

Comment:
A much simpler solution here might be saving/memoizing valInfo2 `saveValidatorsInfo(db, height, valInfo2)` instead of the whole shebang with the blockstore. Thoughts?

Reply:
> A much simpler solution here might be saving/memoizing valInfo2 `saveValidatorsInfo(db, height, valInfo2)` instead of the whole shebang with the blockstore. Thoughts?

Ofcourse, if we can save all validatorInfo.  The previous implement only save whole latest validatorInfo .

Reply:
> A much simpler solution here might be saving/memoizing valInfo2 `saveValidatorsInfo(db, height, valInfo2)` instead of the whole shebang with the blockstore. Thoughts?

so should I modify code to `saveValidatorsInfo(db, height, valInfo2)` ?